code/workspaces/infrastructure-api/src/controllers/projectsController.js

```diff
@@ -82,8 +82,12 @@ const projectDocumentValidator = () => service.middleware.validator([
   check('key')
     .exists()
     .withMessage("'key' must be specified in request body.")
-    .isLength({ min: 2 })
-    .withMessage("'key' must be 2 or more characters long.")
+    .isLength({ min: 2, max: 8 })
+    .withMessage("'key' must be 2-8 characters in length.")
+    .isAlpha()
+    .withMessage("'key' can only contain letters.")
+    .isLowercase()
+    .withMessage("Letters in 'key' must be lowercase.")
     .trim(),
```

Comment:
https://stackoverflow.com/questions/46213401/trim-input-values-before-express-validator-in-node-js suggests trim() should come before any length checks.

Reply:
Good thinking. I have moved `.trim()` up the validation chain to be executed after the `.exists()` check.